### PR TITLE
[feat] fetch and display all cases

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Added Typescript types corresponding to Schema types.
Added all case fetching in cases/page.tsx

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

image

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Review `src/types/schemaTypes` file, then `src/api/supabase/queries/case.ts` then `src/app/cases/page.tsx`.

May need to look out for merge error in `src/api/supabase/queries/case.ts`

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

**Review by rahihazra — APPROVED** [review]
looks so good!!

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// inspiration from https://stackoverflow.com/a/57941711/22063638
+// example timestamp format: 2024-01-18T11:22:40+00:00
+// assumes +00:00
+export default function timestampStringToDate(ts: string) {
```
Comment (pragyakallanagoudar) [inline src/utils/helpers.ts:4]: helper functions NICE

**Review by pragyakallanagoudar — APPROVED** [review]
looks AMAZING!!!! tested it out and everything seems to be working correctly
